onecgiar-pr-server/src/api/results/share-result-request/share-result-request.service.ts

```diff
@@ -94,24 +95,383 @@ export class ShareResultRequestService {
+        (user: { userId: number }) => user.userId,
+      );
+
+      for (const request of shareInitRequests) {
+        const [initOwner, result, , initMembers] =
+          await this.getRequestRelatedData(request, resultId, emailTemplate);
+
+        const receivers = await this.getEmailAndNotificationRecipients(
+          initMembers,
+          emailTemplate,
+          request.shared_inititiative_id,
+          initOwner.id,
         );
-      const requestPendingData =
-        await this._shareResultRequestRepository.getPendingByUser(
-          user.id,
-          role,
+
+        if (!receivers.userNotification.length) {
+          this._logger.warn(
+            `No recipients found for request ID: ${request.share_result_request_id}`,
+          );
+          continue;
+        }
+
+        const matchUsers = receivers.userNotification.filter((userId) =>
+          usersOnlineIds.includes(userId),
+        );
+
+        if (!matchUsers.length) {
+          this._logger.warn(
+            `No online users to notify for request ID: ${request.share_result_request_id}`,
+          );
+          continue;
+        }
+
+        const data = await this._shareResultRequestRepository.findOne({
+          select: this.getRequestSelectFields(),
+          relations: this.getRequestRelations(),
+          where: {
+            result_id: result.id,
+            shared_inititiative_id: request.shared_inititiative_id,
+          },
+        });
+
+        const description = data.is_map_to_toc
+          ? `Request as contributor for result ${result.result_code}`
+          : `Contribution request for result ${result.result_code}`;
+
+        const notification: NotificationDto = {
+          title: 'Contribution request',
+          desc: description,
+          result: data,
+        };
+
+        await this._socketManagementService.sendNotificationToUsers(
+          matchUsers.map(String),
+          notification,
         );
+      }
+    } catch (error) {
+      this._logger.error('Error sending socket notifications', error);
+      this._handlersError.returnErrorRes({ error, debug: true });
+    }
+  }
+
+  private async getRequestRelatedData(
+    request: ShareResultRequest,
+    resultId: number,
+    emailTemplate: string,
+  ) {
+    return await Promise.all([
+      this._clarisaInitiativeRepository.findOne({
+        where: { id: request.owner_initiative_id },
+      }),
+      this._resultRepository.findOne({ where: { id: resultId } }),
+      this._clarisaInitiativeRepository.findOne({
+        where: { id: request.shared_inititiative_id },
+      }),
+      this._roleByUserRepository.find({
+        where: {
+          initiative_id:
+            emailTemplate === EmailTemplate.CONTRIBUTION
+              ? request.shared_inititiative_id
+              : request.owner_initiative_id,
+          role: In([3, 4, 5]),
+          active: true,
+        },
+        relations: ['obj_user'],
```

Comment:
@JuanGuzman-io please note that as per [TypeORM documentation](https://github.com/typeorm/typeorm/blob/e7649d2746f907ff36b1efb600402dedd5f5a499/src/find-options/FindOptionsRelations.ts#L43), this way of declaring the relations (using a `string` array), is going to be deprecated on the next version of the dependency, so please use (for this case) `relations: {obj_user: true}` instead